### PR TITLE
Initial flatpak for org.free_astro.siril

### PR DESCRIPTION
**Pull request by hfiguiere**
Context https://twitter.com/SiriL_Official/status/1219512587773190144

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build org.free_astro.siril

**Comment by hfiguiere** [comment]
Patch submitted upstream https://gitlab.com/free-astro/siril/merge_requests/32

File: org.free_astro.siril.json [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    "command": "siril",
+    "rename-icon": "siril",
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=home",
+        /* Needed for dconf to work */
+        "--filesystem=xdg-run/dconf", "--filesystem=~/.config/dconf:ro",
```
Comment (TingPing) [inline org.free_astro.siril.json]: Remove dconf permissions

File: org.free_astro.siril.json [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        },
+        {
+	    "name": "opencv",
+	    "buildsystem": "cmake-ninja",
+	    "builddir": true,
+	    "cleanup": [
+		"/bin",
+                "/include",
+                "/lib/pkgconfig",
+                "/lib/*.a"
```
Comment (TingPing) [inline org.free_astro.siril.json]:
```suggestion
```

File: org.free_astro.siril.json [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+	    ],
+	    "config-opts": [
+		"-D CMAKE_BUILD_TYPE=RELEASE",
```
Comment (TingPing) [inline org.free_astro.siril.json]: Whitespace nitpicks.

Comment (hfiguiere) [reply org.free_astro.siril.json]: done

File: org.free_astro.siril.json (hunk 1 above)
Comment (hfiguiere) [reply org.free_astro.siril.json]: done

**Comment by hfiguiere** [comment]
Lots of whitespace fixing. I thought Emacs would have had it right...

**Comment by bilelmoussaoui** [comment]
Anyone from upstream interested in (co-)maintaining the repository once created?

**Comment by lock042** [comment]
Many thanks to @hfiguiere for his work. I hope the package is ok now :).

**Comment by lock042** [comment]
@hfiguiere :
If I well undestand the json file, some dep (not mandatory but very recommenced!!!)  are missing:

```
libraw for DSLR RAW files import
libffms2 for films import (any format supported by ffmpeg)
libtiff (>= 4) for TIFF format support
libjpeg or compatible libraries like libjpeg-turbo for JPEG format support
libpng (>= 1.6) for PNG format support
libavformat, libavutil (>= 55.20), libavcodec, libswscale and libswresample for avi export (usually provided by ffmpeg)
libcurl for the Web-based new version check and the astrometry solver
```

Also, `exiv2` is not required for this version.

**Comment by hfiguiere** [comment]
Libraw, libffms2 are there. The others are in the SDK already.

(ok I can remove exiv2).

Here is the output of the configure script on the builder:

```
FB: Running: flatpak build --die-with-parent --env=FLATPAK_BUILDER_BUILDDIR=/run/build/siril --nofilesystem=host --filesystem=/srv/buildbot/worker/build-x86_64-2/build/.flatpak-builder/build/siril-1 --bind-mount=/run/build/siril=/srv/buildbot/worker/build-x86_64-2/build/.flatpak-builder/build/siril-1 --build-dir=/run/build/siril --bind-mount=/run/ccache=/srv/buildbot/worker/build-x86_64-2/build/.flatpak-builder/ccache --unshare=network --env=SOURCE_DATE_EPOCH=1579653910 '--env=CFLAGS=-O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection' '--env=CXXFLAGS=-O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection' '--env=LDFLAGS=-L/app/lib -Wl,-z,relro,-z,now -Wl,--as-needed' --env=CCACHE_DIR=/run/ccache/disabled --env=PATH=/app/bin:/usr/bin --env=LD_LIBRARY_PATH=/app/lib --env=PKG_CONFIG_PATH=/app/lib/pkgconfig:/app/share/pkgconfig:/usr/lib/pkgconfig:/usr/share/pkgconfig --env=FLATPAK_BUILDER_N_JOBS=48 /srv/buildbot/worker/build-x86_64-2/build/.flatpak-builder/rofiles/rofiles-hcvYYG ./configure --prefix=/app
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether NLS is requested... yes
Unescaped left brace in regex is passed through in regex; marked by <-- HERE in m/^(.*)\${ <-- HERE ?([A-Z_]+)}?(.*)$/ at /app/bin/intltool-update line 1065.
Unescaped left brace in regex is passed through in regex; marked by <-- HERE in m/\${ <-- HERE ?AC_PACKAGE_NAME}?/ at /app/bin/intltool-update line 1193.
Unescaped left brace in regex is passed through in regex; marked by <-- HERE in m/\${ <-- HERE ?PACKAGE}?/ at /app/bin/intltool-update line 1194.
Unescaped left brace in regex is passed through in regex; marked by <-- HERE in m/\${ <-- HERE ?AC_PACKAGE_VERSION}?/ at /app/bin/intltool-update line 1195.
Unescaped left brace in regex is passed through in regex; marked by <-- HERE in m/\${ <-- HERE ?VERSION}?/ at /app/bin/intltool-update line 1196.
Unescaped left brace in regex is passed through in regex; marked by <-- HERE in m/\${ <-- HERE ?AC_PACKAGE_NAME}?/ at /app/bin/intltool-update line 1222.
Unescaped left brace in regex is passed through in regex; marked by <-- HERE in m/\${ <-- HERE ?PACKAGE}?/ at /app/bin/intltool-update line 1223.
Unescaped left brace in regex is passed through in regex; marked by <-- HERE in m/\${ <-- HERE ?AC_PACKAGE_VERSION}?/ at /app/bin/intltool-update line 1224.
Unescaped left brace in regex is passed through in regex; marked by <-- HERE in m/\${ <-- HERE ?VERSION}?/ at /app/bin/intltool-update line 1225.
Unescaped left brace in regex is passed through in regex; marked by <-- HERE in m/\${ <-- HERE ?\w+}?/ at /app/bin/intltool-update line 1226.
checking for intltool >= 0.50... 0.51.0 found
checking for intltool-update... /app/bin/intltool-update
checking for intltool-merge... /app/bin/intltool-merge
checking for intltool-extract... /app/bin/intltool-extract
checking for xgettext... /usr/bin/xgettext
checking for msgmerge... /usr/bin/msgmerge
checking for msgfmt... /usr/bin/msgfmt
checking for gmsgfmt... /usr/bin/msgfmt
checking for perl... /usr/bin/perl
checking for perl >= 5.8.1... 5.30.0
checking for XML::Parser... ok
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... gcc3
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for gcc option to support OpenMP... -fopenmp
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking how to run the C++ preprocessor... g++ -E
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking execinfo.h usability... yes
checking execinfo.h presence... yes
checking for execinfo.h... yes
checking sys/vfs.h usability... yes
checking sys/vfs.h presence... yes
checking for sys/vfs.h... yes
checking sys/vmount.h usability... no
checking sys/vmount.h presence... no
checking for sys/vmount.h... no
checking sys/statfs.h usability... yes
checking sys/statfs.h presence... yes
checking for sys/statfs.h... yes
checking sys/statvfs.h usability... yes
checking sys/statvfs.h presence... yes
checking for sys/statvfs.h... yes
checking sys/filio.h usability... no
checking sys/filio.h presence... no
checking for sys/filio.h... no
checking for reallocarray... yes
checking for timegm... yes
checking for gmtime_r... yes
checking for backtrace... yes
checking for log in -lm... yes
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for GTK... yes
checking for FFTW... yes
checking for GSL... no
checking for GSL... yes
checking for CFITSIO... yes
checking for LIBCONFIG... yes
checking whether we are using the GNU C++ compiler... (cached) yes
checking whether g++ accepts -g... (cached) yes
checking dependency style of g++... (cached) gcc3
checking how to run the C++ preprocessor... g++ -E
checking whether g++ supports C++11 features by default... yes
checking for OPENCV... yes
checking for LIBRAW... yes
checking for LIBTIFF... yes
checking for jpeg_mem_src in -ljpeg... yes
checking for LIBPNG... yes
checking for FFMS2... yes
checking for FFMPEG... yes
checking for CURL... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking if compiling for Linux... yes
checking if compiling for Win32... no
checking for Mac OS X platform... no
checking for native Mac OS X... no
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating pixmaps/Makefile
config.status: creating pixmaps/icons/Makefile
config.status: creating platform-specific/Makefile
config.status: creating platform-specific/linux/Makefile
config.status: creating po/Makefile.in
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing po/stamp-it commands
```

**Comment by nedrichards** [comment]
@lock042 a number of these are provided by the shared runtime

**Comment by lock042** [comment]
Yes sorry, I removed my message just before your two answers. Sorry about that.
But about exiv2 it is not needed, I can confirm ;).

It is only for the master branch. By the way, I've tested the build with
`flatpak install --user https://dl.flathub.org/build-repo/13576/org.free_astro.siril.flatpakref`
That looks good!!

**Pull request closed without merging**